### ISSUE TITLE
Iterate over ParamSet

### ISSUE THREAD

**Issue by ThomasAlban** [bevyengine/bevy#11674]
I would love to be able to iterate over a ParamSet rather than being limited to calling each `p0()`, `p1()` ... method. This would help me avoid having to do this:
```rust
for mut visibility in q.p0().p0().iter_mut() { /* ... */ }
for mut visibility in q.p0().p1().iter_mut() { /* ... */ }
for mut visibility in q.p0().p2().iter_mut() { /* ... */ }
for mut visibility in q.p0().p3().iter_mut() { /* ... */ }
// ...
```

**Comment by alice-i-cecile** [comment]
Yep, fully on board with this existing. I suspect that implementing this will require some reflection-style code gen: iterating over fields is not typically supported in a straightforward way.

**Comment by Adamkob12** [comment]
I think this impossible to implement from Bevy's side. Reflection can't work because that would require each of the items being iterated on implementing Reflect. 

You might be able to define your own SystemParam and abstract that away though.

**Comment by james7132** [comment]
Given that ParamSet elements can have different types, this is likely not going to work without some dynamic dispatch or unsafe transmutation. Closing this out as this is likely not really feasible to do, nor are the benefits that clear here.
